### PR TITLE
feat: update documentation and test cases

### PR DESCRIPTION
**Pull request by gritzkoo**
https://coveralls.io/github/gritzkoo/golang-health-checker

### CODE REVIEW

**Review by danilolutz — APPROVED** [review]

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+name: test
```
Comment (joaomantovani) [inline .github/workflows/main.yaml:1]: Is this name `test` correctly?

Comment (joaomantovani) [reply .github/workflows/main.yaml:1]: Isn't the pipeline name?

File: .github/workflows/main.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        redis-version: [4, 5, 6]
+        go: ["1.11", "1.12", "1.13", "1.14", "1.15"]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Memcached Service
```
Comment (joaomantovani) [inline .github/workflows/main.yaml:12]:
```suggestion
      - name: Sets up a Memcached server
```

File: .github/workflows/main.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      - name: Start Redis
+        uses: supercharge/redis-github-action@1.1.0
+        with:
+          redis-version: ${{ matrix.redis-version }}
+      - uses: actions/setup-go@v1
+        with:
+          go-version: ${{ matrix.go }}
+      - run: go test -v -coverprofile=profile.cov ./...
+
+      - name: Send coverage
```
Comment (joaomantovani) [inline .github/workflows/main.yaml:23]: Where is the place to send the coverage?

```suggestion
      - name: Send coverage **to where?**
```

File: .github/workflows/main.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        uses: shogo82148/actions-goveralls@v1
+        with:
+          path-to-profile: profile.cov
+          flag-name: Go-${{ matrix.go }}
+          parallel: true
+
+          # notifies that all test jobs are finished.
```
Comment (joaomantovani) [inline .github/workflows/main.yaml:30]:
```suggestion
          # notifies that all test jobs are finished.
```

Comment (joaomantovani) [reply .github/workflows/main.yaml:30]: This suggestion above removes the empty line

File: README.md [hunk 5]
```diff
@@ -27,42 +55,34 @@ import (
...
+          Type: healthcheck.Memcached,
+          Name: "Memcached server",
+          Host: "localhost",
           Port: "11213",
         }, {
-          Type:    healthcheck.Web,             // this prop will determine the kind of check, the list of types available in structs.go
-          Name:    "Github Integration",        // the name of you integration to display in response
-          Host:    "https://github.com/status", // you can pass host:port and omit Port attribute
-          TimeOut: 5,                           // default value to web call is 10s
-          Headers: []healthcheck.HTTPHeader{ // to customize headers to perform a GET request
```
Comment (joaomantovani) [inline README.md:65]: I think you should take the comments back again. It's very useful

File: README.md [hunk 6]
```diff
@@ -124,21 +141,43 @@ And detailed call will return a JSON as below
 }
 ```
```
Comment (joaomantovani) [inline README.md:143]:
```suggestion
## How to contribute
```

File: docker-compose.yaml [hunk 7]
```diff
@@ -37,5 +37,4 @@ services:
 
 networks:
   default:
-    external:
-      name: gritzkoo-golang-health-checker
+    name: gritzkoo-golang-health-checker
```
Comment (joaomantovani) [inline docker-compose.yaml:40]: Is the spacing correctly?

File: docs/DEVELOPING.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+* [x] Redis
+* [x] Memcache
+* [x] Web API calls
+* [ ] Mongodb
+* [ ] Mysql
+* [ ] Postgres
+* [ ] RabbitMQ
+
+ And any other that in this very moment we just don't known
```
Comment (joaomantovani) [inline docs/DEVELOPING.md:30]:
```suggestion
 There are no plans to add more integrations for now, but if you want to colaborate, open a PR or a issue
```

**Review by joaomantovani — APPROVED** [review]

File: .github/workflows/main.yaml (hunk 1 above)
Comment (gritzkoo) [reply .github/workflows/main.yaml:1]: this name is the one displayed in the generated badge to include in readme.md

File: .github/workflows/main.yaml (hunk 3 above)
Comment (gritzkoo) [reply .github/workflows/main.yaml:23]: send coverage do coverall.io

**Pull request merged**